### PR TITLE
docs: :memo: add description of `extract_resource_properties()`

### PR DESCRIPTION
**Pull request by lwjohnst86**
## Description

These changes add a description for an extraction of properties function for data files. This function needs to be exposed to the user, especially since it needs to be exposed to be used properly for the web app. It also simplifies the `create_resource_properties()` function a lot!

## Reviewer Focus

This PR needs an in-depth review.

### CODE REVIEW

File: docs/design/python-functions.qmd [hunk 1]
```diff
@@ -251,6 +241,22 @@ follows the Frictionless Data specification.
...
 
+### `extract_resource_properties(path, data_path)`
+
+This takes the data found at the `data_path` location and creates a JSON
+object following the Frictionless Data standard. Internally it uses the
+frictionless Python package to guess and fill in some of the property
+fields required by the Frictionless Data resource specification. This
+function is often followed by `edit_resource_properties()` to fill in
+any remaining missing fields and then `verify_resource_properties()`
+function. Usually, you use either this function or the
```
Comment (martonvago) [inline docs/design/python-functions.qmd:252]:
```suggestion
any remaining missing fields and then the `verify_resource_properties()`
function. Usually, you use either this function or the
```

File: docs/design/python-functions.qmd [hunk 2]
```diff
@@ -251,6 +241,22 @@ follows the Frictionless Data specification.
...
+`create_resource_properties()` function to create the resource
+properties for the specific data resource given. The `path` argument
+points to the resource `id` folder the properties are for; use
+`path_resource()` to help give the correct path. to give the path to the
+resource; Outputs a JSON object, use `write_resource_properties()` to
```
Comment (martonvago) [inline docs/design/python-functions.qmd:257]:
```suggestion
`path_resource()` to help give the correct path. Outputs a JSON object,
use `write_resource_properties()` to
```

**Review by martonvago — CHANGES_REQUESTED** [review]
Makes lots of sense!

**Review by martonvago — DISMISSED** [review]

File: docs/design/python-functions.qmd [hunk 3]
```diff
@@ -251,6 +241,22 @@ follows the Frictionless Data specification.
...
+This takes the data found at the `data_path` location and creates a JSON
+object following the Frictionless Data standard. Internally it uses the
+frictionless Python package to guess and fill in some of the property
+fields required by the Frictionless Data resource specification. This
+function is often followed by `edit_resource_properties()` to fill in
+any remaining missing fields and then the `verify_resource_properties()`
+function. Usually, you use either this function or the
+`create_resource_properties()` function to create the resource
+properties for the specific data resource given. The `path` argument
+points to the resource `id` folder the properties are for; use
```
Comment (signekb) [inline docs/design/python-functions.qmd:255]:
```suggestion
points to the resource `id` folder; use
```

File: docs/design/python-functions.qmd [hunk 4]
```diff
@@ -251,6 +241,22 @@ follows the Frictionless Data specification.
...
+`path_resource()` to help give the correct path. Outputs a JSON object,
+use `write_resource_properties()` to
```
Comment (signekb) [inline docs/design/python-functions.qmd:257]:
```suggestion
`path_resource()` to help give the correct path. Outputs a JSON object.
Use `write_resource_properties()` to
```

File: docs/design/python-functions.qmd [hunk 5]
```diff
@@ -251,6 +241,22 @@ follows the Frictionless Data specification.
...
+### `extract_resource_properties(path, data_path)`
+
+This takes the data found at the `data_path` location and creates a JSON
+object following the Frictionless Data standard. Internally it uses the
+frictionless Python package to guess and fill in some of the property
+fields required by the Frictionless Data resource specification. This
+function is often followed by `edit_resource_properties()` to fill in
+any remaining missing fields and then the `verify_resource_properties()`
+function. Usually, you use either this function or the
+`create_resource_properties()` function to create the resource
```
Comment (signekb) [inline docs/design/python-functions.qmd:253]: So, just for my own understanding: Would `create_resoure_properties()` be for when you don't have a data file to extract from?

**Review by signekb — COMMENTED** [review]
Nice! 
Just a few minor language things and a question for my own understanding :)

**Review by K-Beicher — COMMENTED** [review]
Looks sensible to me.

File: docs/design/python-functions.qmd (hunk 5 above)
Comment (lwjohnst86) [reply docs/design/python-functions.qmd:253]: Yea, exactly. Like you know what data will eventually look like, but haven't got it yet and you want to prepare the resource.

Comment (signekb) [reply docs/design/python-functions.qmd:277]: Amazing. Thanks 👍

**Pull request merged**